main.go

```diff
@@ -28,15 +33,58 @@ func main() {
 	}
 	logger.SetLevel(parsedLogLevel)
 
-	// prometheus conf
-	promMetricsHandler, _ := app.CreateMetricsHandler(logger, *namespace, *serviceName, *instanceName)
-	requestMetricName, _ := promMetricsHandler.CreateRequestsMetric()
+	if len(metricNames) == 0 {
+		panic(errors.New("at least one metric name should be given"))
+	}
+
+	// num_of_requests metric must exists since its metric handler contains the logic that makes the server a proxy,
+	// without it requests won't be forwarded to the forwardAddress
+	if !stringInSlice(string(metrics.NumOfRequestsMetricName), metricNames) {
+		metricNames = append(metricNames, string(metrics.NumOfRequestsMetricName))
+	}
+
+	var metricHandlers []metrics.MetricHandler
+	for _, metricName := range metricNames {
+		metricHandler, err := createMetricHandler(metricName, logger, *forwardAddress, *listenAddress, *namespace, *serviceName, *instanceName)
+		if err != nil {
+			panic(err)
+		}
+		metricHandlers = append(metricHandlers, metricHandler)
+	}
```

Comment:
I would opt to move this block and `createMetricHandler ` inside `NewProxyServer`, but just a suggestion (it's not wrong to be here or anything)
Then you could reuse some of the params which are properties of the server and it would clean up `main.go`